alertmanager/mesh.go

```diff
@@ -0,0 +1,137 @@
+// Definitions in this file were largely taken from
+// https://github.com/prometheus/alertmanager/blob/master/cmd/alertmanager/main.go
+//
+// Original copyright notice:
+//
+// Copyright 2015 Prometheus Team
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package alertmanager
```

Comment:
I'd be for only moving it to its own package once it's used from two places, for the sake of keeping all the AM stuff together (it's not really AM-specific yeah, but mesh isn't used by anything else yet). The original Alertmanager even had all this just as part of `main.go` :)